### PR TITLE
add .unit to Execution object

### PR DESCRIPTION
**Pull request by rubanm**

### CODE REVIEW

File: scalding-core/src/main/scala/com/twitter/scalding/Execution.scala [hunk 1]
```diff
@@ -423,6 +423,9 @@ object Execution {
    */
   def fromFuture[T](fn: ConcurrentExecutionContext => Future[T]): Execution[T] = FutureConst(fn)
 
+  /** This creates a constant Execution[Unit] */
+  def unit: Execution[Unit] = from(())
```
Comment (isnotinvain) [inline scalding-core/src/main/scala/com/twitter/scalding/Execution.scala]: could be a val right?

Comment (reconditesea) [inline scalding-core/src/main/scala/com/twitter/scalding/Execution.scala]: .from is a def, what will be benefit of making this a val?

**Comment by johnynek** [comment]
merge when green IMO

File: scalding-core/src/main/scala/com/twitter/scalding/Execution.scala (hunk 1 above)
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/Execution.scala]: If it is a val, then you only have one of them, and don't reallocate them
(that said, Executions are not in the hot path, so no big deal).

On Wed, Feb 4, 2015 at 12:54 PM, Kevin Lin <notifications@github.com> wrote:

> In scalding-core/src/main/scala/com/twitter/scalding/Execution.scala
> <https://github.com/twitter/scalding/pull/1189#discussion_r24128280>:
>
> > @@ -423,6 +423,9 @@ object Execution {
> >     */
> >    def fromFuture[T](fn: ConcurrentExecutionContext => Future[T]): Execution[T] = FutureConst(fn)
> >
> > +  /** This creates a constant Execution[Unit] */
> > +  def unit: Execution[Unit] = from(())
>
> .from is a def, what will be benefit of making this a val?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twitter/scalding/pull/1189/files#r24128280>.
>



-- 
Oscar Boykin :: @posco :: http://twitter.com/posco

**Pull request merged**